### PR TITLE
Weekly PR

### PR DESCRIPTION
**Pull request by EugeneButusov**
+ Fix #131 ConcurrentModificationException in LevelUp

### CODE REVIEW

**Comment by EugeneButusov** [comment]
@borisspektor @refaelos, it's ready to merge.

File: SoomlaAndroidLevelUp/src/com/soomla/levelup/LevelUp.java [hunk 1]
```diff
@@ -207,7 +207,7 @@ private static void findInternalLists(HashMap<String, JSONObject> objects, Strin
         try {
             List<String> classes = Arrays.asList(listClasses);
             for (JSONObject objectJSON : objects.values()) {
-                findInternalLists(objects, classes, listName, objectJSON);
+                findInternalLists(new HashMap<String, JSONObject>(objects), classes, listName, objectJSON);
```
Comment (borisspektor) [inline SoomlaAndroidLevelUp/src/com/soomla/levelup/LevelUp.java]: @EugeneButusov this is not a good solution. You are creating a new HashMap where the values will be stored, while they need to be stored in the original HashMap.
The solution should be in the line before, where you're traversing the list. `objects.values()` should be cloned.

Comment (vedi) [inline SoomlaAndroidLevelUp/src/com/soomla/levelup/LevelUp.java]: @EugeneButusov, I guess @borisspektor means something like that:
```
List<String> classes = Arrays.asList(listClasses);
Set<xxx> valuesSet = new HashSet<>(objects.values());
for (JSONObject objectJSON : valuesSet) {
    findInternalLists(objects, classes, listName, objectJSON);
```

Comment (borisspektor) [inline SoomlaAndroidLevelUp/src/com/soomla/levelup/LevelUp.java]: @vedi @EugeneButusov exactly.
Need to test to make sure that it solves the exception.

**Pull request merged**